src/coffee/extend/ForceDirectedGraph.coffee

```diff
@@ -0,0 +1,154 @@
+Function::property = (prop, desc) ->
+  Object.defineProperty @prototype, prop, desc
+
+
+# Hacky Edge implementation
+class Edge
+  constructor: (source, target) ->
```

Comment:
I see. Then I think either `Graph.addEdge(P, P)` or using an `Edge` class will make sense. Changing Pair to accept references seems a bit more dangerous :)